programs/pkey/gen_key.c

```diff
@@ -321,7 +321,7 @@ int main(int argc, char *argv[])
         }
     } else
 #endif /* MBEDTLS_RSA_C */
-#if defined(MBEDTLS_ECP_C)
+#if defined(MBEDTLS_ECP_C) && !defined(MBEDTLS_USE_PSA_CRYPTO)
```

Comment:
I'm glad you were not happy about this, because it becomes unnecessary once we undo the change to `ecp.h` :)